lumen/app/Console/Commands/ImportEvents.php

```diff
@@ -20,12 +20,22 @@ public function __construct(EventDataProvider $eventDataProvider) {
     }
 
     public function handle() {
+        $importedItemsCount = 0;
         $location = $this->argument('location');
-        
-        $events = $this->eventDataProvider->getByLocation($location);
 
-        foreach ($events as $event) {
-            dispatch(new \App\Jobs\ImportEvent($event));
+        $this->info("Starting Import of event in $location");
+        $this->info('Getting Info to start to import');
+        $paginationMetadata = $this->eventDataProvider->getPaginationMetadataByLocation($location);
+
+        for ($page = 1; $page <= $paginationMetadata->page_count; $page++) {
```

Comment:
it's fine, we could improve this using a `do while` or even return an iterator from the data provider to handle the iteration over the pagination.